pymc/logprob/abstract.py

```diff
@@ -131,6 +131,11 @@ def _icdf_helper(rv, value, **kwargs):
 class MeasurableVariable(abc.ABC):
```

Comment:
Oh we are not subclassing from it directly so its not being used (almost all of our logprob rewrites should have failed).

We usually subclass from the Ops... so maybe we should add a class decorator to add these properties to init and pass the remaining ones to the original Op class when it's initialized.

Reply:
The decorator would be applied on the Measurable subclasses of those Ops, e.g.

```python
@measurable_op
class MeasurableElemwise(Elemwise):
  pass
```

So that we can kind of subclass without multiple inheritance. The decorator would override `__init__` to capture the new parameters we need, and call the original `__init__` with the rest.

Reply:
But maybe Multiple Inheritance is the best here. We might need to think/try both to figure it out.

Reply:
I tried the decorator method and it works fine. There is one problem that whenever the ```MeasurableOp``` class is called in the ```find_measurable_op``` method, in many cases(almost all), it requires some default values for all the parameters including the metainfo parameters and the original init parameters. The ```get_default_measurable_metainfo``` does not work on ```node.op``` as ```node.op``` is an object of the ```Op``` and not ```MeasurableOp``` where the parameters are added. 
For eg:
https://github.com/pymc-devs/pymc/blob/a617bf29393200b9b998ae354863eda4d0df670c/pymc/logprob/binary.py#L44-L97
Replacing line 93 and 94 as 
```python
ndim_supp, supp_axes, measure_type = get_default_measurable_metainfo(node.op)
compared_op = MeasurableComparison(ndim_supp, supp_axes, measure_type, node_scalar_op)
```
produces an attribute error 
```
AttributeError: 'Elemwise' object has no attribute 'ndim_supp'
```
I guess multiple Inheritance will also lead to this error as the metainfo are passed to our ```MeasurableOps``` and not the original pytensor ```Ops```.

Reply:
I think that's exactly why multiple inheritance would work better, as those attributes would now be part of the Op through the MeasurableClass path.

Reply:
I have made attributes part of the Op through the MeasurableClass path by including them in the ```init``` a done according to the first commit which , I believe is the approach that multiple inheritance follows. The attributes are accessible through an object of the MeasurableClass and not the Op class. However, I am still confused on how to proceed with default values. As mentioned earlier, ```node.op``` is derived from the MeasurableClass itself and not the Op. Hence in case of MeasurableElemwise and sub-class which derive from it, ```scalar_op``` being an attribute of the op itself gives no error for ```node.op.scalar_op``` but ```node.op.ndim_supp``` does.